hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/MessageFlyweight.java

```diff
@@ -175,7 +175,11 @@ public String getStringUtf8() {
     }
 
     public Data getData() {
-        return new HeapData(getByteArray());
+        final int length = buffer.getInt(index + offset);
+        index += Bits.INT_SIZE_IN_BYTES;
+        Data data = new ByteBufferHeapData(buffer.byteArray(), index + offset, length);
```

Comment:
Is this `ByteBufferHeapData` going to be stored in Hazelcast data structures, IMap, ICache etc?